### ISSUE TITLE
integer overflow in 'core.shmmem' output

### ISSUE THREAD

**Issue by juliusf** [kamailio/kamailio#2824]
Hi,

we are using kamailio with more than 4GB of shared memory. When using 'core.shmem' we are expierencing an integer overflow in the output. 

The issue can be found here:

https://github.com/kamailio/kamailio/blob/1ddc27f199061025a6a43da3e8a1388fcaddfe19/src/modules/corex/corex_rpc_shm.c#L72

I currently don't have the time to setup a build environment so I can't fix it myself.

cheers.

**Comment by seabra** [comment]
A sexta, 13/08/2021, 10:38, Julius Flohr ***@***.***>
escreveu:

**Comment by miconda** [comment]
The rcp supported int and double types, the values being stored over an int field. I added a few more conversion types for long/long long variants, storing them over the double field. The standard jsonrpc defines anyhow on 'number' (double) as a type.

binrpc had a further hack for double fields, being serialized as integer. I changed to serialize as long long.

kamctl didn't need any change as it prints the numbers coming in the formatted json. kamcmd print the double values as long long if they have no decimals.

Overall, the current master should print properly the shm stats with values over 4G. If not, open a new issue.

There were many changes, practically the rpc framework got support for new data types, therefore won't be backported.
